turbinia/processors/mount_local.py

```diff
@@ -51,15 +51,35 @@ def PreprocessMountDisk(evidence):
 
   evidence.mount_path = tempfile.mkdtemp(prefix='turbinia', dir=mount_prefix)
 
-  if hasattr(evidence, 'mount_partition') and evidence.mount_partition:
-    src_path = '{0:s}-part{1:d}'.format(
-        evidence.local_path, evidence.mount_partition)
-  else:
-    src_path = evidence.local_path
+  # Two steps: first use losetup to parse the RawDisk eventual partition table
+  # and make block devices per partitions.
+  # Then we mount the partition provideed, or the first one we find.
 
   # TODO(aarontp): Remove hard-coded sudo in commands:
   # https://github.com/google/turbinia/issues/73
-  mount_cmd = ['sudo', 'mount', src_path, evidence.mount_path]
+  losetup_command = [
+      'sudo', 'losetup', '--show', '--nooverlap', '--find', '-P',
+      evidence.local_path]
+  log.info('Running command {0:s}'.format(' '.join(losetup_command)))
+  try:
+    losetup_device = subprocess.check_output(losetup_command)
+  except subprocess.CalledProcessError as e:
+    raise TurbiniaException('Could not set losetup devices {0!s}'.format(e))
+  evidence.losetup_device = losetup_device.strip()
+
+  # Here, evidence.losetup_device is something like '/dev/loopX'.
+
+  path_to_partition = '{0:s}p{1:d}'.format(
```

Comment:
I know the mount_partition will normally be set to 1 because you changed the default arguments, but we will potentially have other clients that can make turbinia requests (e.g. dfTimewolf), and the default value here is None.  Can you make an explicit check here for None and then set it to 1 in case this happens?

Reply:
SGTM